modules/core/src/main/java/org/apache/synapse/aspects/newstatistics/StatisticsEntry.java

```diff
@@ -0,0 +1,510 @@
+	 * @param isResponse    is message context belong to an response
+	 */
+	public synchronized void createLog(String componentId, ComponentType componentType, int msgId, String parentId,
+	                                   long startTime, boolean isResponse) {
+		if (openLogs.isEmpty()) {
+			StatisticsLog statisticsLog = new StatisticsLog(componentId, componentType, msgId, -1, -1, "", startTime);
+			messageFlowLogs.add(statisticsLog);
+			openLogs.addFirst(messageFlowLogs.size() - 1);
+			if (log.isDebugEnabled()) {
+				log.debug("Starting statistic log at root level [ElementId|" + componentId +
+				          "]|[MsgId|" + msgId + "]");
+			}
+		} else {
+			if (openLogs.getFirst() == 0) {
+				if (messageFlowLogs.get(0).getComponentId().equals(componentId)) {
+					if (log.isDebugEnabled()) {
+						log.debug("Statistics event is ignored as it is a duplicate of root " + "element");
+					}
+					return;
+				}
+			}
+			if (parentId.equals("")) {
+				int parentIndex = getFirstLogWithMsgId(msgId);
+				createNewLog(componentId, componentType, msgId, parentIndex, startTime, isResponse);
+			} else {
+				int parentIndex = getComponentIndex(parentId, msgId);
+				if (parentIndex > -1) {
+					createNewLog(componentId, componentType, msgId, parentIndex, startTime, isResponse);
+				} else {
+					if (log.isDebugEnabled()) {
+						log.debug("Invalid stating element");
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Create statistics log at the start of a fault sequence
+	 *
+	 * @param componentId   componentId of the statistic reporting element
+	 * @param componentType component Type of the statistic reporting element
+	 * @param msgId         msgId of the statistics reporting message context
+	 * @param parentId      parentId of the reporting statistic event
+	 * @param startTime     starting time of the statistics reporting event
+	 * @param isResponse    is message context belong to an response
+	 */
+	public synchronized void createFaultLog(String componentId, ComponentType componentType, int msgId, String parentId,
+	                                        Long startTime, boolean isResponse) {
+		int parentIndex = getParentForFault(parentId, msgId);
+		createNewLog(componentId, componentType, msgId, parentIndex, startTime, isResponse);
+		addFaultsToParents(parentIndex);
+		openFaults += 1;
+	}
+
+	/**
+	 * close a opened statistics log after all the statistics collection relating to that statistics
+	 * component is ended
+	 *
+	 * @param componentId componentId of the statistic event sender
+	 * @param msgId       message id of the message context
+	 * @param parentId    parent of the statistic event sender
+	 * @param endTime     endTime of the statistics event
+	 * @return true if there are no open message logs in openLogs List
+	 */
+	public synchronized boolean closeLog(String componentId, int msgId, String parentId, long endTime) {
+		int componentLevel;
+		if (parentId.equals("")) {
```

Comment:
I have removed returning empty strings and checking for them in the PR. Instead I use null values to do the same.